### PR TITLE
feat: color Grid3D by coordinates.

### PR DESCRIPTION
**Pull request by LeonidPolukhin**
Grid3D can be colored by a coordinate value and the corresponding option is added to the layers API.

### CODE REVIEW

File: typescript/packages/subsurface-viewer/src/layers/grid3d/grid3dLayer.stories.tsx [hunk 1]
```diff
@@ -15,15 +15,16 @@ import {
...
 export default {
     component: SubsurfaceViewer,
     title: "SubsurfaceViewer/Grid3D",
-} as ComponentMeta<typeof SubsurfaceViewer>;
+} as Meta<typeof SubsurfaceViewer>;
 
 type NumberQuad = [number, number, number, number];
 
-const Template: ComponentStory<typeof SubsurfaceViewer> = (args) => (
+const Template: StoryFn<typeof SubsurfaceViewer> = (args) => (
```
Comment (t0oF-azpn) [inline typescript/packages/subsurface-viewer/src/layers/grid3d/grid3dLayer.stories.tsx:27]: Keep the old deprecated way. I'm currently updating that more globally, using the StoryObj approach...

File: typescript/packages/subsurface-viewer/src/layers/grid3d/grid3dLayer.ts [hunk 2]
```diff
@@ -85,6 +85,16 @@ async function load_data(
     return Promise.all([points, polys, properties]);
 }
 
+/**
+ * Enumerates possible coloring modes.
+ */
+export enum TGrid3DColoringMode {
```
Comment (t0oF-azpn) [inline typescript/packages/subsurface-viewer/src/layers/grid3d/grid3dLayer.ts:91]: Does this escalate to top level index ?
You can run `npm run doc` to see the warnings...

File: typescript/packages/subsurface-viewer/src/layers/grid3d/privateGrid3dLayer.ts [hunk 3]
```diff
@@ -73,6 +75,8 @@ export interface privateLayerProps extends ExtendedLayerProps {
 
 const defaultProps = {
     colorMapName: "",
+    colorMapClampColor: [200, 200, 200],
```
Comment (t0oF-azpn) [inline typescript/packages/subsurface-viewer/src/layers/grid3d/privateGrid3dLayer.ts:78]: Magic numbers ?!

**Review by t0oF-azpn — CHANGES_REQUESTED** [review]

File: typescript/packages/subsurface-viewer/src/layers/grid3d/grid3dLayer.stories.tsx (hunk 1 above)
Comment (LeonidPolukhin) [reply typescript/packages/subsurface-viewer/src/layers/grid3d/grid3dLayer.stories.tsx]: Reverted.

File: typescript/packages/subsurface-viewer/src/layers/grid3d/grid3dLayer.ts (hunk 2 above)
Comment (LeonidPolukhin) [reply typescript/packages/subsurface-viewer/src/layers/grid3d/grid3dLayer.ts:91]: enum TGrid3DColoringMode is now proporgated to the topmost index and the corresponding doc generated.

File: typescript/packages/subsurface-viewer/src/layers/grid3d/privateGrid3dLayer.ts (hunk 3 above)
Comment (LeonidPolukhin) [reply typescript/packages/subsurface-viewer/src/layers/grid3d/privateGrid3dLayer.ts:78]: No magic here, just light grey color

File: typescript/packages/subsurface-viewer/src/layers/grid3d/grid3dLayer.ts [hunk 4]
```diff
@@ -104,6 +114,12 @@ export interface Grid3DLayerProps extends ExtendedLayerProps {
      */
     propertiesData: string | number[] | Float32Array;
 
+    /**
+     * Defines how the cells are to be colored:
+     * by property value or by a coordinate.
+     */
+    coloringMode: TGrid3DColoringMode;
```
Comment (hkfb) [inline typescript/packages/subsurface-viewer/src/layers/grid3d/grid3dLayer.ts:121]: Maybe this should be optional, since "Property" is a sensible default.

**Review by hkfb — CHANGES_REQUESTED** [review]

**Review by hkfb — APPROVED** [review]

**Pull request merged**